lib/response.ex

```diff
@@ -12,16 +18,43 @@ defmodule HTTPServer.Response do
 
   @carriage_return "\r\n"
 
-  def send_resp(status_code, body, headers) do
+  def send_resp(resp) do
+    %{resp | headers: Headers.collect_headers(resp.headers)}
+  end
+
+  def build(req = %Request{method: "HEAD"}, status_code, res_body, media_type) do
+    resp =
+      new()
+      |> status(status_code)
+      |> body(res_body)
+      |> Headers.build(req, media_type)
+
+    send_resp(%{resp | body: ""})
+  end
+
+  def build(req, status_code, res_body, media_type) do
+    new()
+    |> status(status_code)
+    |> body(res_body)
+    |> Headers.build(req, media_type)
+    |> send_resp()
+  end
+
+  def new() do
     %__MODULE__{
-      status_code: status_code,
-      status_message: status_message(status_code),
+      status_code: nil,
+      status_message: nil,
       resource: "HTTP/1.1",
-      headers: Headers.collect_headers(headers),
-      body: body
+      headers: Headers.new(),
+      body: nil
     }
   end
```

Comment:
Is this really a "new"? is there a better name?  Should `new/0` be a function that returns something else? (like what this actually is?)

Reply:
At one point I was calling it `empty_res/0`, which does feel like it describes the function better than `new/0` does at the moment!

Reply:
It may be fine as is.  I just think this is so common in Elixir there would be an existing pattern for it.  How does it differ then just using `%Request{}`

Reply:
Just updated this so Response.build and Headers.build start with `%__MODULE__{}` instead of `new/0`, and adds `resource/1` and `headers/3` as explicit functions in the pipeline!